zrml/authorized/src/lib.rs

```diff
@@ -60,36 +63,63 @@ mod pallet {
         <<T as Config>::MarketCommons as MarketCommonsPalletApi>::MarketId;
     pub(crate) type MomentOf<T> = <<T as Config>::MarketCommons as MarketCommonsPalletApi>::Moment;
 
+    pub type CacheSize = ConstU32<64>;
+
     #[pallet::call]
     impl<T: Config> Pallet<T> {
         /// Overwrites already provided outcomes for the same market and account.
         #[frame_support::transactional]
-        #[pallet::weight(T::WeightInfo::authorize_market_outcome())]
+        #[pallet::weight(T::WeightInfo::authorize_market_outcome(
+            CacheSize::get(),
+            CacheSize::get()
+        ))]
         pub fn authorize_market_outcome(
             origin: OriginFor<T>,
             market_id: MarketIdOf<T>,
             outcome: OutcomeReport,
-        ) -> DispatchResult {
+        ) -> DispatchResultWithPostInfo {
             T::AuthorizedDisputeResolutionOrigin::ensure_origin(origin)?;
             let market = T::MarketCommons::market(&market_id)?;
             ensure!(market.status == MarketStatus::Disputed, Error::<T>::MarketIsNotDisputed);
             ensure!(market.matches_outcome_report(&outcome), Error::<T>::OutcomeMismatch);
-            match market.dispute_mechanism {
-                MarketDisputeMechanism::Authorized => {
-                    AuthorizedOutcomeReports::<T>::insert(market_id, outcome);
-                    Ok(())
-                }
-                _ => Err(Error::<T>::MarketDoesNotHaveDisputeMechanismAuthorized.into()),
-            }
+            ensure!(
+                market.dispute_mechanism == MarketDisputeMechanism::Authorized,
+                Error::<T>::MarketDoesNotHaveDisputeMechanismAuthorized
+            );
+
+            let ids_len_1 = Self::remove_auto_resolve(&market_id);
+            let now = frame_system::Pallet::<T>::block_number();
+            let correction_period_ends_at = now.saturating_add(T::CorrectionPeriod::get());
+            let ids_len_2 =
+                T::DisputeResolution::add_auto_resolve(&market_id, correction_period_ends_at)?;
+
+            let report = AuthorityReport { resolve_at: correction_period_ends_at, outcome };
+            AuthorizedOutcomeReports::<T>::insert(market_id, report);
+
+            Ok(Some(T::WeightInfo::authorize_market_outcome(ids_len_1, ids_len_2)).into())
         }
     }
 
     #[pallet::config]
     pub trait Config: frame_system::Config {
+        /// The period in which the authority has to report.
+        #[pallet::constant]
+        type ReportPeriod: Get<Self::BlockNumber>;
+
         /// Event
         type Event: From<Event<Self>> + IsType<<Self as frame_system::Config>::Event>;
 
-        /// Market commons
+        /// The period, in which the authority can correct the outcome of a market.
+        #[pallet::constant]
+        type CorrectionPeriod: Get<Self::BlockNumber>;
```

Comment:
It's just that the **final decision** could rather have a fat-finger protection. But, okay, if you would not like a correction period, this would simplify the protocol.

Reply:
I'm not opposed to it, just looking for a good argument to have it. Usually, there are no do-overs on the blockchain.

But comparing this to other MDMs, I guess there's always the option to do another dispute. In court, for example, it's not a problem if a juror fat-fingers because they can correct their choice (if they notice the mistake) and other jurors will still vote correctly. There's no such safety net for `authorized` without the correction period.

Since you already did the work, let's keep the correction period around.